### DISCUSSION TITLE
"Abort fetching component for route" error

### DISCUSSION THREAD

**Discussion by Zerebokep** [vercel/next.js#39040] [Help] [answered]
I receive the following error:

`Error: Abort fetching component for route: "/"`

using router.push on my registration site, after setting a user object to my auth context

`setUser(user)

router.push('/')`

Anyone has an idea what is causing this error?

**Comment by MrJoy** [comment]
I've struggled with this a bit.  I'm not 100% certain that I understand the issue but it seems to be the case that this sort of problem arises if I call `router.push` from the main rendering flow.  I.E. if I hoist such calls into `useEffect` (or other event handler), the problem seems to go away.  That's consistent with how such things are demonstrated here: https://nextjs.org/docs/api-reference/next/router

**Reply by Zerebokep** [reply]
Doesn't help on my end unfortunately. :/

**Reply by MrJoy** [reply]
I wound up needing a bit more than this myself.  Because of multiple dependencies, I'm getting my useEffect body called multiple times.  Calling `router.push` / `router.replace` multiple times in the same page view / whatever definitely causes things to explode.  So I basically started using `useState` to track when I was calling `router.<whatever>` and ensuring I didn't do so twice.  That's completely fixed the issue for me -- or so it seems.

**Reply by Zerebokep** [reply]
Oh wow, didn't thought of that, that could also be the reason in my code. Thanks for letting me know mate, appreciate it. :)

**Reply by Shnupel** [reply]
hi, this error appears when we create an event and don't clear it. Therefore, if we create setTimeout/setInterval or an event handler in UseEffect, we will get an error. To solve it, you need to remove the event handler when itMount or setInterval / setTimeout

**Reply by MrJoy** [reply]
@Shnupel Are you saying that `router.push` creates such an event?  If so, how would we get a reference to the event so we can clear it?  Or are you saying this is another case that can introduce such an error?

**Reply by gioimtg2003** [reply]
> hi, this error appears when we create an event and don't clear it. Therefore, if we create setTimeout/setInterval or an event handler in UseEffect, we will get an error. To solve it, you need to remove the event handler when itMount or setInterval / setTimeout

thank you

**Comment by Zerebokep** [comment, marked as answer]
Solved the problem as suggested by @MrJoy and @Shnupel in the comments (not calling router.push multiple times). However, a router.isChangingRoute property would be way better. Created a feature request for it here: https://github.com/vercel/next.js/issues/39202

**Comment by gabeodame** [comment]
Solved with useSession and useEffect on secured pages

`
// imports.......

const {data: session} = useSession()

useEffect(()=> {
      if(session !== undefined && session.id){
         signIn()
    }
})
`

**Comment by youssef-devx** [comment]
Solved it by checking is I already pushed a route.
Solution from _https://stackoverflow.com/questions/73343986/next-js-abort-fetching-component-for-route-login_

**Comment by BasVerdoorn** [comment]
## issue explained
router.push() can sometimes run multiple times, which can prevent it from loading the page completely before loading the next router.push() page.

## stack overflow solution
stack overflow answer where they handle multiple routers.push(): [click here](https://stackoverflow.com/a/75872313/21563418)

## idea developers 
maybe it is an idea to solve this problem by default or with an option at the router.

maby you could try to find a way to only load the last route without causing an error and decreasing the speed or make it an option to disable the routing untill the route is fully loaded. You can look at the stack overflow issue. 

## link to the same question
[same question](https://github.com/vercel/next.js/issues/46329#issuecomment-1510912462)
